pkgs/development/libraries/science/math/magma/default.nix

```diff
@@ -1,48 +1,82 @@
-{ lib, stdenv, fetchurl, cmake, gfortran, ninja, cudaPackages, libpthreadstubs, lapack, blas }:
+{ lib
+, stdenv
+, fetchurl
+, cmake
+, ninja
+, gfortran
+, libpthreadstubs
+, lapack
+, blas
+, cudatoolkit
```

Comment:
please stick with the package set, cudaPackages. Using separate expressions risks creating an invalid configuration. Whenever we have subsets we should always pass in the set.

Reply:
This is already done here: https://github.com/NixOS/nixpkgs/pull/206995/commits/4f9a886672a622f01bff9890c7ded2fa45d660bc#diff-ab5748dc9567516fefba8344056b51ec1866adeace380f46e58a7af3d619ea22R36123

Reply:
Unless you want me to explicitly specify ``cudaPackages`` here?

Reply:
There are also several other packages in all-packages.nix doing the exact same thing I'm doing here, so I'm not sure why we would want to specify the entire cudaPackages in the magma derivation when we can do the same in all-packages.nix and be more granular.

Reply:
If you can't mix and match versions of packages in a set (as I assume cudaPackages are) it's better to take the whole set. That way people can't accidentally override one cuda dependency with a package from a different cuda set and so on.

Reply:
It's not more expensive or anything

Reply:
Okay, now that makes sense.
cudaPackages is a slightly special case here because of the broad incompatibilities between versions.
I'll change it.